src/server/network.ts

```diff
@@ -476,3 +540,15 @@ export function mergeHeaders(headers: (types.HeadersArray | undefined | null)[])
     result.push({ name: lowerCaseToOriginalCase.get(lower)!, value });
   return result;
 }
+
+function filterEmpties(obj: { [k: string]: string|number|undefined } | undefined) {
+  if (obj)
+    return Object.entries(obj).filter(([_, v]) => v !== null && v !== undefined && (typeof v === 'number' ? v >= 0 : v !== '')).reduce((o, [prop, val]) => { const merged = (o || {}); merged[prop] = val; return merged; }, undefined as { [k: string]: string|number|undefined }|undefined);
```

Comment:
This grew organically and should be re-written as a simple for-loop for readability. Will do w/ follow up change.